renderdoc/driver/d3d12/d3d12_device_wrap.cpp

```diff
@@ -2132,10 +2132,183 @@ void WrappedID3D12Device::CopyDescriptorsSimple(UINT NumDescriptors,
   }
 }
 
+template <typename SerialiserType>
+bool WrappedID3D12Device::Serialise_OpenSharedHandle(SerialiserType &ser, HANDLE NTHandle, REFIID riid, void **ppvObj)
+{
+  ID3D12DeviceChild *devChild = ser.IsWriting() ? (ID3D12DeviceChild *)*ppvObj : NULL;
```

Comment:
This cast isn't valid because the `void*` coming in is `ID3D12Resource*`. It might work by coincidence but it's a danger for the future. Unfortunately I don't think ther's a better solution than just switching on `riid` up here and casting to the appropriate type to fetch the ID.